### PR TITLE
Five sections - Inheritance

### PR DESCRIPTION
**Pull request by Olehqa**

### CODE REVIEW

File: src/main/java/homeworks/fivesection/inheritance/Human.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package homeworks.fivesection.inheritance;
+
+public abstract class Human {
+    public abstract String getBio();
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/inheritance/Human.java]: follow ordering, first variable, constructor, only after function

File: src/main/java/homeworks/fivesection/inheritance/Employee.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    private String firstName;
+    private String secondName;
+    private String profession;
+    private double salary;
+
+    public Employee() {
+    }
+
+    public Employee(String firstName , String secondName , String profession) {
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/inheritance/Employee.java]: ?

File: src/main/java/homeworks/fivesection/inheritance/Employee.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        this.firstName = firstName;
+    }
+
+    public void setSecondName(String secondName) {
+        this.secondName = secondName;
+    }
+
+    public String getBio() {
+
+        return String.format ( "The bio should not so long for \n , \n" , getFirstName () , getSecondName () );
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/inheritance/Employee.java]: be

File: src/main/java/homeworks/fivesection/inheritance/InheritanceMain.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    public static void main(String[] args) {
+        Human hum = new Student ( "Manu" , "Pkiao" );
+        Human hum2 = new Employee ( "Alice" , "Kimble" , "oldProfession" );
+
+        Human humanAbs = new Human () {
+            @Override
+            public String getBio() {
+                return null;//for what it should be used
+            }
+        };
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/inheritance/InheritanceMain.java:24]: try to use variable humanAbs and call function getBio

File: src/main/java/homeworks/fivesection/inheritance/Student.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
+package homeworks.fivesection.inheritance;
+
+import com.sun.javafx.binding.StringFormatter;
+
+public final class Student extends Human {
+
+    private String firstName;
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/inheritance/Student.java]: remove them,

File: src/main/java/homeworks/fivesection/inheritance/Student.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    private String nativeLanguage;
+    private String speciality;
+    private String groupName;
+    private int course;
+
+    public Student(String firstName , String secondName) {
+    }
+
+    public Student(String firstName , String secondName , String speciality , String groupName , int course) {
+    }
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/inheritance/Student.java]: add here realization with using constructor from Human ,
super( with parameters)

File: src/main/java/homeworks/fivesection/inheritance/InheritanceMain.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package homeworks.fivesection.inheritance;
+
+public class InheritanceMain {
+
+    public static void main(String[] args) {
+        Human hum = new Student ( "Manu" , "Pkiao" );
+        Human hum2 = new Employee ( "Alice" , "Kimble" , "oldProfession" );
+
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/inheritance/InheritanceMain.java:17]: add usage of getBio()

File: src/main/java/homeworks/fivesection/inheritance/Student.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+import com.sun.javafx.binding.StringFormatter;
+
+public final class Student extends Human {
+
+    private String firstName;
+    private String secondName;
+    private String nativeLanguage;
+    private String speciality;
+    private String groupName;
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/inheritance/Student.java:6]: add log4j library

**Review by iikarak — CHANGES_REQUESTED** [review]

File: src/main/java/homeworks/fivesection/inheritance/AbstractHuman.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+/*
+ * AbstractHuman
+ *
+ * 12.06.2020
+ */
+
+public abstract class AbstractHuman {
+
+    private String firstName;
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/inheritance/AbstractHuman.java]: update access modifier

File: src/main/java/homeworks/fivesection/inheritance/Employee.java [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+    public Employee(String firstName , String secondName , String profession , double salary) {
+        super ( firstName , secondName );
+        this.profession = profession;
+        this.salary = salary;
+    }
+
+    @Override
+    public String getFirstName() {
+        return super.getFirstName ();
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/inheritance/Employee.java]: Usless unless you add some additional implementation here

File: src/main/java/homeworks/fivesection/inheritance/Employee.java [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+            return false;
+        }
+
+        if (getClass () != otherObject.getClass ()) {
+            return false;
+        }
+
+        Employee other = (Employee) otherObject;
+
+        return Objects.equals ( getFirstName () , other.getFirstName () ) && Objects.equals ( getLastName () , other.getLastName () );
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/inheritance/Employee.java]: use equals of class from variable not Objects.equals

File: src/main/java/homeworks/fivesection/stringutils/StringUtils.java [hunk 12]
```diff
@@ -0,0 +1,50 @@
+package homeworks.fivesection.stringutils;
+
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+import java.util.regex.Pattern;
+
+public class StringUtils {
+
+    public static boolean isValidByOWASP(String email) {
+        String emailRegex = "^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*@" +
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/stringutils/StringUtils.java:8]: Lets do this simplier:
before dog, we have also symbols, also not less than 1, and with out some special characters; 
contains '@' domain- several symbols, not less than one, after it we should have dot, after is subdomain.

File: src/main/java/homeworks/fivesection/stringutils/StringUtils.java [hunk 13]
```diff
@@ -0,0 +1,50 @@
+package homeworks.fivesection.stringutils;
+
+import javax.mail.internet.AddressException;
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/stringutils/StringUtils.java]: remove this lib

File: src/main/java/homeworks/fivesection/stringutils/StringUtils.java [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+import javax.mail.internet.InternetAddress;
+import java.util.regex.Pattern;
+
+public class StringUtils {
```
Comment (iikarak) [inline src/main/java/homeworks/fivesection/stringutils/StringUtils.java:5]: move this class to framework

**Review by iikarak — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 15]
```diff
@@ -25,7 +25,16 @@
                 </configuration>
             </plugin>
         </plugins>
+
+
     </build>
+    <dependencies>
```
Comment (Olehqa) [inline pom.xml:29]: REMOVE

**Pull request merged**